project4/3700crawler

```diff
@@ -134,21 +174,56 @@ class WebCrawler:
             exit(1)
         self.session = match.group(1)
 
-        self.set_cookie()
-
-        login_resp = self.POST(LOGIN_FORM, make_formstr({
+        login_resp = self.POST(LOGIN_FORM, LOGIN_FORM, make_formstr({
             'username': self.username,
             'password': self.password,
             'csrfmiddlewaretoken': self.csrf,
             'submit': 'Log in',
-            'next': '/fakebook/'
+            'next': r'%2Ffakebook%2F'
```

Comment:
I'm unfamiliar with this syntax, sorry. What do the `r` and `%2` do?

Reply:
Oh! r denotes it as a raw string, which means to not perform any sort of escaping inside the string. I don't remember why I though the string had to be raw, I think that my editor was highlighting %2F and sleepy me thought it was an escape sequence.

The %2F is the HTML sanitized sequence for a '/' character